test/tests.spec.ts

```diff
@@ -713,6 +713,61 @@ describe("node-saml /", function () {
         });
       });
 
+      it("supports 'onlyIfPresent' option for validateInResponseTo - does not validate when not present", async () => {
```

Comment:
IMHO This test description does not "add up". Why should result be non-valid (or "Exact match Invalid Signature, as inResponseTo should not reject" as it says at the end of testcase) if Response is missing `inResponseTo` fields and configuration is `onlyIfPresent`?

I bet this test case's SAML response was copy paste from some other test case and it used to have validly signed assertion and it - SAML response - was modified for this test case so that `inResponseTo` attributes were removed from response and assertion (leaving assertion's signature invalid). I.e. what this test case seems to be testing is that when `options.validateInResponseTo` has `truthy` value `onlyIfPresent` that this particular code block is bypassed successfully (which already has sort of "onlyIfPresent" functionality due xpath search returning possibly `falsy` or empty array IF there didn't happen to be any Reponse element with inResponseTo attrtibute at any namespace - `passport-saml`/`node-saml` is not using namespaces in xpath queries):
https://github.com/node-saml/node-saml/blob/269ea7c8f1abe9ede03d4c9228d5a150013e9e46/src/saml.ts#L748-L757
and function fails at this point:
https://github.com/node-saml/node-saml/blob/269ea7c8f1abe9ede03d4c9228d5a150013e9e46/src/saml.ts#L783-L785
i.e. it never tests this functionality (whis is located in `processValidlySignedAssertionAsync` which is executed after assertion signature validation):
https://github.com/node-saml/node-saml/blob/269ea7c8f1abe9ede03d4c9228d5a150013e9e46/src/saml.ts#L1109-L1139

As a randon side note (note related to this pull request): Just in case if you wonder - while you are browsing the code: `passport-saml`'s/ǹode-samls signature validation is a bit of a "mess". As long as Response has either (validly signed assertion) OR (validly signed top level signature AND validly signed assertion) attacker is free to alter content of Response's inResponseTo, issuer etc values any way as he/she wants (i.e. "outer" inReponse value can always be controlled by attacker under previously mentioned conditions). If SAML authentication response has only one signature and if that signature is associated to Response then and only then outer inResponseTo value is also covered by `passport-saml`/`node-saml` XML signature validation (along with the rest of the SAML authn response). I.e. this particular testcase happens to test (due to how XML response which is used as test material is signed) only parts of the message which can be modified by attacker (without getting caught by such action).  Luckily it seems that `passport-saml`/`node-saml` tries to verify that IF there was `inResponseTo` value at Response level (i.e. if attacker didn't remove it completely) and IF assertion level `inResponseTo` is not equal with Response level `inResponseTo` then response is discarded due obvious "response modification attempt". 

Background (definitely not related to this pull request):
Quite a few other SP implementations provide configuration options to make missing or invalid top level signature a "hard error". I.e. if person who is configuring `passport-saml`/`node-saml` powered SP knows that his/her IdP always provide top level signature he/she can configure service provider discard SAML authentication responses which doesn't have valid top level signatures. For more background information see e.g. https://github.com/node-saml/passport-saml/discussions/671#discussioncomment-2136937

Reply:
You got it - this was a copy/paste from another test case. I don't know all the inner workings of this library - I only really understand the assertion piece that was discussed in https://github.com/node-saml/passport-saml/issues/329. I therefore did not feel confident writing a test from scratch - I simply wanted to cover the change in behavior using a test case that appeared to already cover an end-to-end scenario. Open to suggestions on changing it